### PR TITLE
Book store API

### PR DESCRIPTION
**Pull request by shaqayq**
In this part of the project, I use API for the project to store data and fetch it again.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,31 +1,81 @@
...
-});
+// export const removeBook = (id) => ({
+//   type: REMOVE_BOOK,
+//   id,
+// });
+
+// export const fetching = (data) => ({
+//   type: FETCH_NEW_BOOK,
+//   data,
+// });
```
Comment (vechicin) [inline src/redux/books/books.js]: Please remove this line of commented code to keep a cleaner codebase.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,31 +1,81 @@
...
+    method: 'POST',
+    body: JSON.stringify(book),
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+
+  const data = await response.text();
+  return data;
+});
```
Comment (vechicin) [inline src/redux/books/books.js]: When I click the `Add Button` the book is not added. Please fix this.

File: src/redux/books/books.js [hunk 3]
```diff
@@ -1,31 +1,81 @@
...
+export const deleteBook = createAsyncThunk(REMOVE_BOOK, async (id) => {
+  await fetch(`${BaseURL}/id`, {
+    method: 'DELETE',
+    body: JSON.stringify({
+      item_id: id,
+    }),
+  });
+
+  return id;
+});
```
Comment (vechicin) [inline src/redux/books/books.js]: When I click on the `Remove` button, the books are removed from the UI. However, after I refresh the page they are rendered again. The books should be permanently removed from the API after user interaction, please fix this.

File: src/components/Home.js [hunk 4]
```diff
@@ -1,10 +1,17 @@
...
+  const lists = useSelector((state) => state.books) || [];
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getBooks());
+  }, []);
+
   return (
     <>
       <div className="bookList">
```
Comment (vechicin) [inline src/components/Home.js:17]: I'm seeing this error in the console.

Screen Shot 2022-08-19 at 8 25 38

Please fix it.

**Review by vechicin — CHANGES_REQUESTED** [review]
# Hi @shaqayq ✋, 

Good job so far! 🔥 
There are some issues that you still need to work on to go to the next project but you are almost there! 🚨 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights ⚡ 

- [x] Linters checks are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 📂 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Home.js (hunk 4 above)
Comment (shaqayq) [reply src/components/Home.js:17]: @vechicin !
Thank you for reviewing my code and for your feedback about it. I'm going to fix all this. :blush:

File: src/redux/books/books.js (hunk 1 above)
Comment (DeliceLydia) [reply src/redux/books/books.js]: Addressed

File: src/redux/books/books.js (hunk 3 above)
Comment (DeliceLydia) [reply src/redux/books/books.js]: Resolved

File: src/components/Home.js (hunk 4 above)
Comment (DeliceLydia) [reply src/components/Home.js:17]: fixed

**Review by DeliceLydia — APPROVED** [review]
## Hi @shaqayq, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**